### PR TITLE
Add config syntax support for 'transport=telnet'

### PR DESCRIPTION
**Pull request by The-Loeki**
I just ran into this issue & hacked up this PR :) 

Fixes https://github.com/napalm-automation/napalm-ios/issues/132

### CODE REVIEW

**Review by mirceaulinic — APPROVED** [review]
This looks good to me, but I will defer it to @ktbyers for final call.

File: napalm_ios/ios.py
Comment (dbarrosop) [inline napalm_ios/ios.py:76]: please, call the `optional_arg`: `ios_transport` to avoid collisions with other drivers.

**Comment by dbarrosop** [comment]
Could you also document this optional_arg here, please?

https://github.com/napalm-automation/napalm/blob/master/docs/support/index.rst#list-of-supported-optional-arguments

Something like:

```
transport (ios, eos) - Transport to use when connecting to the device.  Supported transports are the ones supported by the underlying library. Defaults to https on EOS and ssh on IOS.
```

Thanks! :)

File: napalm_ios/ios.py
Comment (ktbyers) [reply napalm_ios/ios.py:76]: @dbarrosop FYI, here you recommended calling it `transport` and the PR implemented it that way.

https://github.com/napalm-automation/napalm-ios/issues/132

IMO we should call it `transport` as it for any netmiko-based driver we will have the same pattern.

Comment (dbarrosop) [reply napalm_ios/ios.py:76]: Ok, sorry about that. Let's call it transport. I will send a PR to update the `eos` driver.

Comment (dbarrosop) [reply napalm_ios/ios.py:76]: I updated my comment regarding the documentation.

**Comment by mirceaulinic** [comment]
Postponed this for 0.8.0 as per @ktbyers' suggestion

**Comment by matejv** [comment]
`is_alive` method needs updating as well.

When using telnet [`remote_conn`](https://github.com/napalm-automation/napalm-ios/blob/develop/napalm_ios/ios.py#L182) will be a `telnetlib.Telnet` instance that doesn't have a `transport` attribute. I'm not sure what is the right way to check if a telnet connection is active.

**Comment by dbarrosop** [comment]
Note there are CI issues we have to solve before merging.

**Comment by ktbyers** [comment]
I will update this. Hopefully this week or next week.

**Comment by austind** [comment]
@ktbyers Let me know if you need any help testing the PR

**Comment by ktbyers** [comment]
This change is incorporated here:

https://github.com/napalm-automation/napalm-ios/pull/192

**Comment by austind** [comment]
Successfully ran `get_facts()` against two telnet devices. Thank you!

**Comment by ktbyers** [comment]
@austind Nice!

**Pull request closed without merging**
